otherlibs/win32unix/unix.ml

```diff
@@ -837,11 +842,19 @@ let make_process_env env =
     env;
   String.concat "\000" (Array.to_list env) ^ "\000"
 
+let create_process_aux prog args optenv fd1 fd2 fd3 =
+  let fd1 = dup ~cloexec:false fd1 in
+  let fd2 = dup ~cloexec:false fd2 in
+  let fd3 = dup ~cloexec:false fd3 in
+  let res = win_create_process prog args optenv fd1 fd2 fd3 in
+  close fd1; close fd2; close fd3;
+  res
```

Comment:
If dup, win_create_process or close releases the runtime lock or invokes the GC then another thread could run before the final close. If that thread then also calls create_process_aux the duplicated FDs will leak.

I'm assuming dup and close will be save here but win_create_process probably allocates something. So I agree with @bobot about the race condition. I think the dup should be moved into the win_create_process C stub and run with the runtime lock held. This would still race with non-ocaml threads forking at the same time but it's the best you can do.